src/LogicItLab/Salesforce/MapperBundle/WsdlValidator.php

```diff
@@ -0,0 +1,208 @@
+
+    private function getClassNameFromFile($file)
+    {
+        $realPath = $file->getRealpath();
+        $fileName = str_replace($this->srcPath, '', $realPath);
+        $className = str_replace('.php', '', $fileName);
+
+        if(strpos($this->srcPath, 'test') !== false) {
+            $className = "Tests/$className";
+        }
+
+        return str_replace('/', '\\', $className);
+    }
+
+    private function filterNonExistentClasses(array $classNames)
+    {
+        $classNames = array_filter($classNames, function ($className) {
+            return class_exists($className);
+        });
+
+        return $classNames;
+    }
+
+    private function hasObject($fileHandler, $objectName): bool
+    {
+        $objectOpeningTag = sprintf('<complexType name="%s">', $objectName);
+        do {
+            $line = trim(fgets($fileHandler));
+            if (strpos($line, 'complexType name') !== false && strcmp($line, $objectOpeningTag) > 0) {
+                fseek($fileHandler, -strlen($line) - 1, SEEK_CUR);
+                return false;
+            }
+        } while (strcmp($line, $objectOpeningTag) != 0 && !feof($fileHandler));
+
+        if (feof($fileHandler)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private function hasField($fileHandler, $fieldName): bool
+    {
+        $fieldOpeningTag = sprintf('<element name="%s"', $fieldName);
+        do {
+            $line = trim(fgets($fileHandler));
+            if (strpos($line, $fieldOpeningTag) !== 0) {
+                if (strpos($line, '</complexType>')) {
+                    return false;
+                }
+                if (strpos($line, 'element name') !== false && strcmp($line, $fieldOpeningTag) > 0) {
+                    fseek($fileHandler, -strlen($line) - 1, SEEK_CUR);
+                    return false;
+                }
+            }
+        } while (strpos($line, $fieldOpeningTag) !== 0 && !feof($fileHandler));
+
+        if (feof($fileHandler)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private function getFieldNames(?array $mapFieldAnnotation): ?array
+    {
+        if(!$mapFieldAnnotation) {
+            return null;
+        }
+
+        $fieldNames = [];
+        foreach ($mapFieldAnnotation as $propertyName => $annotation) {
+            $fieldNames[] = $annotation->name;
+        }
+
+        return $this->sort($fieldNames);
+    }
+
+    /**
+     * @param $fileHandler
+     * @return bool
+     * @throws FileHandlerNotInitializedException
+     */
+    private function initializeFileHandler($fileHandler): void
+    {
+        $sObjectOpeningTag = sprintf('<complexType name="%s">', 'sObject');
+        do {
+            $line = trim(fgets($fileHandler));
+        } while (strcmp($line, $sObjectOpeningTag) !== 0 && !feof($fileHandler));
+
+        if (feof($fileHandler)) {
+            throw new FileHandlerNotInitializedException();
+        }
+    }
+
+    private function sort($array)
```

Comment:
It's only used in one place. We can probably just use the function instead of creating this wrapper for it :)